File: src/main/java/com/sunset/lavafishing/block/BlockEntity/BlockEntityPrometheusBounty.java
package com.sunset.lavafishing.block.BlockEntity;

import com.sunset.lavafishing.util.RegistryCollection.BlockEntityTypeCollection;
import net.minecraft.core.BlockPos;
import net.minecraft.network.chat.Component;
import net.minecraft.world.level.block.entity.ChestBlockEntity;
import net.minecraft.world.level.block.state.BlockState;

import javax.annotation.Nonnull;

public class BlockEntityPrometheusBounty extends ChestBlockEntity
{
    public BlockEntityPrometheusBounty(BlockPos pos, BlockState state) {
        super(BlockEntityTypeCollection.BLOCK_ENTITY_PROMETHEUS_BOUNTY.get(), pos, state);
    }

    @Override
    @Nonnull
    public Component getDefaultName() {
        return Component.translatable(this.getBlockState().getBlock().getDescriptionId());
    }
}

File: src/main/java/club/redux/sunset/lavafishing/block/blockentity/BlockEntityPrometheusBounty.kt
package club.redux.sunset.lavafishing.block.blockentity

import club.redux.sunset.lavafishing.registry.ModBlockEntityTypes
import net.minecraft.core.BlockPos
import net.minecraft.network.chat.Component
import net.minecraft.world.level.block.entity.ChestBlockEntity
import net.minecraft.world.level.block.state.BlockState

class BlockEntityPrometheusBounty(
    pos: BlockPos,
    state: BlockState,
) : ChestBlockEntity(ModBlockEntityTypes.PROMETHEUS_BOUNTY.get(), pos, state) {
    override fun getDefaultName(): Component = Component.translatable(this.blockState.block.descriptionId)
}
